sbi/utils/get_nn_models.py

```diff
@@ -290,25 +296,19 @@ def likelihood_nn(
 
 
 def classifier_nn(
-    model,
-    prior: torch.distributions.Distribution,
-    x_o: torch.Tensor,
-    hidden_features: int = 50,
+    model, theta_dim: int, x_o_dim: int, hidden_features: int = 50,
 ) -> torch.nn.Module:
     """Neural classifier
 
     Args:
         model: Model, one of linear / mlp / resnet
-        prior: Prior distribution
-        x_o: Observation
+        theta_dim: event shape of the prior, number of parameters.
+        x_o_dim: number of elements in a single data point.
```

Comment:
dimensionality of a single data point? Maybe add the shape minus the leading dimension, to be even more specific.

Reply:
We often use `theta_dim` for the "number of parameters" or "dimensionality of the simulator". Same for `x_o_dim`. 

What about calling it `theta_size`  and `x_o_size` because this is how it is used in the context of constructing the nn below, right?

Reply:
Do you mean something like `x.shape[1:]`? 

And I should add it in the docstring explanation or in the code?

Reply:
> We often use `theta_dim` for the "number of parameters" or "dimensionality of the simulator". Same for `x_o_dim`.

Do you not see potential for confusion in `theta_dim` for a [3x3] matrix being 9? or it should be 2?

Dimensionality in Torch/numpy, length of the shape, with  leading batch dimension would be 3. Dimensionality of the single example would be 2. Number of elements would be 9. But in math, depending on the application, dimensionality would also be 9.

I think we need to define `dimension` and `dimensionality` [here](https://github.com/mackelab/sbi/blob/master/sbi/user_input/user-input-checks.md) so that we can consistently refer to it and not get confused every time.

The use of dimension there seems to be according to Torch/numpy conventions: a torch of shape `[3,4,5]` has three dimensions (it says). Now we need the vocabulary to talk about shape[i] (dimensionality?) and also about `shape[1:].sum()` (number of batch elements?). 

Let's discuss this in person.